Sources_v3/Controllers/ChannelController.swift

```diff
@@ -59,7 +59,13 @@ public typealias ChannelController = ChannelControllerGeneric<DefaultDataTypes>
 ///
 public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, DelegateCallable {
     /// The ChannelQuery this controller observes.
-    public let channelQuery: ChannelQuery<ExtraData>
+    public private(set) var channelQuery: ChannelQuery<ExtraData> {
+        didSet {
+            guard oldValue.cid != channelQuery.cid else { return }
+            _channelObserver.reset()
+            _messagesObserver.reset()
```

Comment:
When I think about it now, I think we need to re-initialize the observer immediately so the delegates keep receiving callbacks, no? We should add a test for this.

Reply:
Right, thanks!
For some reason I was sure that `Cached` property will recompute the value. I checked implementation before usage but probably `reset` function name confused me after.